api/types.proto

```diff
@@ -713,3 +727,31 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this references.
+	string name = 1;
+
+	// SecretValueID represents the ID of the specific Secret version that we're
+	// referencing. This identifier exists so that SecretReferences don't leak
+	// any information about the secret contents.
+	string secret_value_id = 2 [(gogoproto.customname) = "SecretValueID"];
```

Comment:
Could you elaborate what the renaming would entail?

Does this involve accepting a `SecretSpec` and updating it on an existing secret?  If so, then unless we enforce that the secret data itself does not change, we lose the previous version of that secret data.  Are we going to take the name the user gives us, and synthesize a new `SecretSpec` ourselves with the same data but the new name?